### PR TITLE
refactor(build-tools): Reorganize build-tools files in preparation for splitting out fluid-build

### PR DESCRIPTION
**Pull request by tylerbutler**
This change does some internal organization within the build-tools package to better isolate fluid-build-only functionality from shared functionality. Everything that is fluid-build only - with the exception of the config and the "getConfig" function - should be within the fluidBuild folder and only referenced from other files within that folder.

This change also:

- Splits the FluidRepo class and the build config into separate files.
- Moves some util functions that are used only within fluid-build or tasks into the fluid-build-specific taskUtils.
- Removes the unused function `resolveNodeModule` from common/utils.ts

**There is no intended behavior change. It's only meant to reorganize existing code.**

There are still some circular relationships that need to be rationalized. For example, the `MonoRepo` and `Package` classes expect the fluid-build properties to be included in package.json, but that config should really be specific to fluid-build. That's not addressed in this change, but it and similar issues will be addressed over time, starting with the introduction of a shared build infrastructure package in #22347.

### CODE REVIEW

File: build-tools/packages/build-tools/src/common/utils.ts [hunk 1]
```diff
@@ -150,6 +126,9 @@ function printExecError(
 	}
 }
 
+/**
+ * @deprecated No replacement.
+ */
 export function resolveNodeModule(basePath: string, lookupPath: string) {
```
Comment (tylerbutler) [inline build-tools/packages/build-tools/src/common/utils.ts:132]: Unused.

File: build-tools/packages/build-tools/src/common/utils.ts [hunk 2]
```diff
@@ -33,35 +31,13 @@ export function getExecutableFromCommand(command: string) {
...
-}
-
 export const statAsync = util.promisify(fs.stat);
 export const lstatAsync = util.promisify(fs.lstat);
 export const readFileAsync = util.promisify(fs.readFile);
 export const writeFileAsync = util.promisify(fs.writeFile);
 export const unlinkAsync = util.promisify(fs.unlink);
 export const existsSync = fs.existsSync;
-export const appendFileAsync = util.promisify(fs.appendFile);
+// export const appendFileAsync = util.promisify(fs.appendFile);
```
Comment (tylerbutler) [inline build-tools/packages/build-tools/src/common/utils.ts:40]: Delete.

**Review by jason-ha — APPROVED** [review]
I agree with PR description - intent to have no behavior change. But there are two changes to use async functions where a synchronous version was used before. Recommend splitting those out.
Would also be nice to note in description that `resolveNodeModule` was removed as unused. (I think everything else was relocated.)

File: build-tools/packages/build-tools/src/fluidBuild/tasks/taskUtils.ts [hunk 3]
```diff
@@ -23,7 +26,7 @@ export function getEsLintConfigFilePath(dir: string) {
 
 export async function getInstalledPackageVersion(packageName: string, cwd: string) {
 	const resolvedPath = require.resolve(packageName, { paths: [cwd] });
-	const packageJsonPath = await lookUpDirSync(resolvedPath, (currentDir) => {
+	const packageJsonPath = await lookUpDirAsync(resolvedPath, async (currentDir) => {
 		return existsSync(path.join(currentDir, "package.json"));
 	});
```
[the remark this answers is not in the document]
Comment (tylerbutler) [reply build-tools/packages/build-tools/src/fluidBuild/tasks/taskUtils.ts:31]: Very good point. I'll split these async changes out as you suggested.

File: build-tools/packages/build-tools/src/common/utils.ts [hunk 4]
```diff
@@ -33,35 +31,12 @@ export function getExecutableFromCommand(command: string) {
...
 export const statAsync = util.promisify(fs.stat);
 export const lstatAsync = util.promisify(fs.lstat);
 export const readFileAsync = util.promisify(fs.readFile);
 export const writeFileAsync = util.promisify(fs.writeFile);
 export const unlinkAsync = util.promisify(fs.unlink);
 export const existsSync = fs.existsSync;
-export const appendFileAsync = util.promisify(fs.appendFile);
 export const realpathAsync = util.promisify(fs.realpath.native);
 export const symlinkAsync = util.promisify(fs.symlink);
 export const mkdirAsync = util.promisify(fs.mkdir);
```
[the remark this answers is not in the document]
Comment (tylerbutler) [reply build-tools/packages/build-tools/src/common/utils.ts:42]: Yeah, this is definitely something I want to clean up. I always thought that this was a node10-ism, and that the promises APIs weren't introduced until later. But [the docs say they were stable in Node10](https://nodejs.org/docs/latest-v10.x/api/fs.html#fs_fs_promises_api), so I can't explain why they were written that way. @curtisman Do you remember?

File: build-tools/packages/build-tools/src/fluidBuild/tasks/taskUtils.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (tylerbutler) [reply build-tools/packages/build-tools/src/fluidBuild/tasks/taskUtils.ts]: Reverted all the unintended async changes.

File: build-tools/packages/build-tools/src/fluidBuild/tasks/taskUtils.ts [hunk 5]
```diff
@@ -23,7 +26,7 @@ export function getEsLintConfigFilePath(dir: string) {
 
 export async function getInstalledPackageVersion(packageName: string, cwd: string) {
 	const resolvedPath = require.resolve(packageName, { paths: [cwd] });
-	const packageJsonPath = await lookUpDirSync(resolvedPath, (currentDir) => {
+	const packageJsonPath = lookUpDirSync(resolvedPath, (currentDir) => {
```
Comment (jason-ha) [inline build-tools/packages/build-tools/src/fluidBuild/tasks/taskUtils.ts:29]: a useless `await` is being removed.

**Review by jason-ha — APPROVED** [review]

**Pull request merged**